client/client.go

```diff
@@ -238,14 +238,22 @@ func (c *authenticatingClient) sendAuthRequest(
 // object-store and container service types have no versions. For these services, the
 // caller may pass "" for apiVersion, to use the service catalogue URL without any
 // version discovery.
-func (c *authenticatingClient) MakeServiceURL(serviceType, apiVersion string, parts []string) (string, error) {
+func (c *authenticatingClient) MakeServiceURL(serviceType, apiVersion string, parts []string) (returnURL string, _ error) {
 	if !c.IsAuthenticated() {
 		return "", errors.New("cannot get endpoint URL without being authenticated")
 	}
+	logger := logging.FromCompat(c.logger)
 	serviceURL, ok := c.serviceURLs[serviceType]
 	if !ok {
 		return "", errors.New("no endpoints known for service type: " + serviceType)
 	}
+
+	defer func() {
+		if returnURL != "" {
+			logger.Tracef("MakeServiceURL: %s",returnURL)
+		}
+	} ()
+
```

Comment:
Might be nice to include the URL you are falling back to in logger.Warningf("falling back to catalogue service URL").

Reply:
We do print out the catalogue near the top of bootstrap... and the url will be printed too if trace is enable.